### PR TITLE
Remove unneeded constructCSRHostName

### PR DESCRIPTION
**Pull request by howardjohn**
I am not 100% sure this is not needed, but Limin mentioned it was and it
would make some other things simpler. Hopefully either a test fails or a
security reviewer rejects this if we do need it...

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: security/pkg/nodeagent/cache/secretcache.go [hunk 1]
```diff
@@ -846,22 +846,7 @@ func (sc *SecretCache) generateSecret(ctx context.Context, token string, connKey
...
-		csrHostName, err = constructCSRHostName(sc.configOptions.TrustDomain, token)
-		if err != nil {
-			cacheLog.Warnf("%s failed to extract host name from jwt: %v, fallback to SDS request"+
-				" resource name: %s", logPrefix, err, connKey.ResourceName)
-			csrHostName = connKey.ResourceName
-		}
-	}
-	cacheLog.Debugf("constructed host name for CSR: %s", csrHostName)
 	options := pkiutil.CertOptions{
-		Host:       csrHostName,
```
Comment (liminw) [inline security/pkg/nodeagent/cache/secretcache.go:864]: I am not sure if we can completely remove the field from CSR, or should fill it with a dummy value (like we do for GCE VM case). John, can you verify that istiod is issuing certificate correctly with this change? @JimmyCYJ @myidpt Do we have any integration tests covering this code path?

Comment (howardjohn) [reply security/pkg/nodeagent/cache/secretcache.go:864]: @JimmyCYJ @myidpt?

**Comment by howardjohn** [comment]
@istio/wg-security-maintainers ping? Any plans to review PRs from people not in the WG?

File: security/pkg/nodeagent/cache/secretcache.go (hunk 1 above)
Comment (liminw) [reply security/pkg/nodeagent/cache/secretcache.go:860]: I asked @JimmyCYJ. We probably only have unit test in place. John, could you manually verify that with the change, istiod is still issuing certificate correctly?

Comment (howardjohn) [reply security/pkg/nodeagent/cache/secretcache.go:860]: What am I supposed to verify? Of course mTLS between pods works, our testing isn't so bad that we don't test that

**Pull request merged**